edge-hub/src/Microsoft.Azure.Devices.Routing.Core/endpoints/StoringAsyncEndpointExecutor.cs

```diff
@@ -44,7 +44,7 @@ public class StoringAsyncEndpointExecutor : IEndpointExecutor
             this.options = Preconditions.CheckNotNull(options);
             this.messageStore = messageStore;
             this.sendMessageTask = Task.Run(this.SendMessagesPump);
-            this.prioritiesToFsms = new AtomicReference<SortedDictionary<uint, EndpointExecutorFsm>>(new SortedDictionary<uint, EndpointExecutorFsm>());
+            this.prioritiesToFsms = new AtomicReference<ImmutableDictionary<uint, EndpointExecutorFsm>>(ImmutableDictionary<uint, EndpointExecutorFsm>.Empty);
```

Comment:
Were we relying on the sorted-ness of the `SortedDictionary`?

Reply:
Yeah, to process messages in priority order.. but he is sorting the dict before using below, so that should work.